realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsDynamicTests.java

```diff
@@ -20,34 +20,484 @@
+        }
+
+        final DynamicRealmObject cat3 = dynamicRealm.where(Cat.CLASS_NAME).equalTo(Cat.FIELD_NAME, "cat3").findFirst();
+        final RealmResults<DynamicRealmObject> cat3Owners = cat3.linkingObjects(Owner.CLASS_NAME, Owner.FIELD_CAT);
+        assertNotNull(cat3Owners);
+        assertTrue(cat3Owners.isEmpty());
+    }
+
+    @Test
+    public void linkingObjects_LIST() {
+        //           source100          source200        source300
+        //            //    \\          \\ || //
+        //        target1   target2     target2
+        //
+        //  /  = object ref
+        //  // = list ref
+        realm.executeTransaction(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                final AllJavaTypes target1 = realm.createObject(AllJavaTypes.class, 1L);
+                final AllJavaTypes target2 = realm.createObject(AllJavaTypes.class, 2L);
+                final AllJavaTypes target3 = realm.createObject(AllJavaTypes.class, 3L);
+
+                final AllJavaTypes source100 = realm.createObject(AllJavaTypes.class, 100L);
+                source100.getFieldList().add(target1);
+                source100.getFieldList().add(target2);
+
+                // list contains three target2s
+                final AllJavaTypes source200 = realm.createObject(AllJavaTypes.class, 200L);
+                source200.getFieldList().add(target2);
+                source200.getFieldList().add(target2);
+                source200.getFieldList().add(target2);
+            }
+        });
+
+        final DynamicRealmObject target1 = dynamicRealm.where(AllJavaTypes.CLASS_NAME).equalTo(AllJavaTypes.FIELD_ID, 1L).findFirst();
+        final DynamicRealmObject target2 = dynamicRealm.where(AllJavaTypes.CLASS_NAME).equalTo(AllJavaTypes.FIELD_ID, 2L).findFirst();
+        final DynamicRealmObject target3 = dynamicRealm.where(AllJavaTypes.CLASS_NAME).equalTo(AllJavaTypes.FIELD_ID, 3L).findFirst();
+
+        // tests sources of target1
+        final RealmResults<DynamicRealmObject> target1Sources = target1.linkingObjects(AllJavaTypes.CLASS_NAME, AllJavaTypes.FIELD_LIST);
+        assertNotNull(target1Sources);
+        assertEquals(1, target1Sources.size());
+        assertEquals(AllJavaTypes.CLASS_NAME, target1Sources.first().getType());
+        assertEquals(100L, target1Sources.first().getLong(AllJavaTypes.FIELD_ID));
+        assertTrue(target1Sources.first().getList(AllJavaTypes.FIELD_LIST).contains(target1));
+        assertTrue(target1Sources.first().getList(AllJavaTypes.FIELD_LIST).contains(target2));
+        assertFalse(target1Sources.first().getList(AllJavaTypes.FIELD_LIST).contains(target3));
+
+        // tests sources of target2
+        final RealmResults<DynamicRealmObject> target2Sources = target2.linkingObjects(AllJavaTypes.CLASS_NAME, AllJavaTypes.FIELD_LIST);
+        assertNotNull(target2Sources);
+        // TODO Is ↓ an expected behavior?
```

Comment:
Yes, Backlinks are exposed as a `RealmResults` so you can use `distinct` on it if you want to only get unique references.